src/rail/estimation/algos/deepdisc.py

```diff
@@ -195,26 +195,17 @@ def run(self):
                 file_path = os.path.join(self.temp_dir, filename)
                 np.save(file_path, reformed_image)
 
-                this_image_metadata = file_path
+                this_image_metadata["filename"] = file_path
 
-        num_gpus = self.config.num_gpus
```

Comment:
These variables were only used in the `launch(...)` call. So instead of defining extra variables, just use the `self.config...` values directly.